test/unit/array_test.rb

```diff
@@ -2,22 +2,37 @@
 
 class ArrayTest < ActiveSupport::TestCase
   context '#to_xml method' do
+    setup do
+      @version = Gem::Version.new(RUBY_VERSION)
+      @ruby_24 = Gem::Version.new('2.4.0')
+    end
+
     should 'not add the attr params as default' do
       xml = Nokogiri::XML::Document.parse([1,2].to_xml)
 
-      assert xml.xpath("//fixnums[@type]").empty?
+      if @version < @ruby_24
+        assert xml.xpath("//fixnums[@type]").empty?
```

Comment:
Can we turn this into a fixed parameter defined in `#setup`?

```ruby
def setup
  num_type = Gem::Version.new(RUBY_VERSION) < Gem::Version.new('2.4.0') ? 'fixnums' : 'integers'
  @num_type_xpath = "//#{num_type}[@type]"
end
...
should ... do
  ...
  assert xml.xpath(@num_type_xpath).empty?
end
```

...unless the idea here is to make it easy to remove in the next iteration.

Reply:
Yes the idea is to upgrade